domain/src/main/java/org/oppia/domain/question/QuestionDataController.kt

```diff
@@ -0,0 +1,83 @@
+package org.oppia.domain.question
+
+import android.content.Context
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import org.oppia.app.model.Question
+import org.oppia.domain.topic.TEST_TOPIC_ID_0
+import org.oppia.domain.topic.TEST_TOPIC_ID_1
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.data.DataProviders
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val QUESTION_DATA_PROVIDER_ID = "QuestionDataProvider"
+
+/** Controller for retrieving an exploration. */
+@Singleton
+class QuestionDataController @Inject constructor(
+  private val questionProgressController: QuestionProgressController,
+  private val dataProviders: DataProviders
+) {
+
+  /**
+   * Returns a list of [Question] objects given a topic ID.
+   */
+  fun getQuestionsForTopic(topicId: String): LiveData<AsyncResult<List<Question>>> {
+    val dataProvider = dataProviders.createInMemoryDataProviderAsync(QUESTION_DATA_PROVIDER_ID) {
+      retrieveQuestionsForTopic(topicId)
+    }
+    return dataProviders.convertToLiveData(dataProvider)
+  }
+
+  /**
+   * Begins a question training session given a list of questions. This method is not expected to fail.
+   * [QuestionProgressController] should be used to manage the play state, and monitor the load success/failure of
+   * the training session.
+   *
+   * Questions will be shuffled and then the training session will begin.
+   *
+   * @return a one-time [LiveData] to observe whether initiating the play request succeeded.
+   * The training session may still fail to load, but this provides early-failure detection.
+   */
+  fun startQuestionTrainingSession(questionsList: List<Question>): LiveData<AsyncResult<Any?>> {
```

Comment:
I noticed this late, but shouldn't this take a list of skill IDs? I expect the controller to take a list of skill IDs and generate a list of questions from that. Please sync with me if my assumption is wrong, otherwise please update this method accordingly.

Reply:
Ah okay got it, I have changed the methods to be this way